### PR TITLE
[improvement] Bring support to configure resident application's outbound provisioning

### PR DESCRIPTION
**Pull request by JayaShakthi97**
**Add outbound provisioning configuration**

https://github.com/wso2/identity-apps/assets/37938529/e18f414f-b079-4596-895f-0ecd635c7d4f

**Update and Delete outbound provisioning configuration**

https://github.com/wso2/identity-apps/assets/37938529/6c4f9d33-811b-4e99-a0ae-1b8d8d4c5a6d

### Related Issues
- https://github.com/wso2/product-is/issues/19100

### Related PRs
- https://github.com/wso2/carbon-identity-framework/pull/5474

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/provisioning/api/outbound-provisioning.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { AxiosError, AxiosRequestConfig, AxiosResponse } from "axios";
+import { OutboundProvisioningConfigurationInterface } from "../../applications/models/application";
+import { store } from "../../core/store";
+
+/**
+ * Get an axios instance.
+ */
+const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
+    .httpRequest.bind(AsgardeoSPAClient.getInstance())
+    .bind(AsgardeoSPAClient.getInstance());
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/api/outbound-provisioning.ts:30]: Binding once is enough

File: apps/console/src/features/provisioning/api/use-get-resident-outbound-provisioning-connectors.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AxiosRequestConfig } from "axios";
+import { HttpMethods } from "modules/core/src/models";
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/api/use-get-resident-outbound-provisioning-connectors.ts:20]: Use @wso2is/core

File: apps/console/src/features/core/configs/routes.tsx [hunk 3]
```diff
@@ -1680,6 +1680,20 @@ export const getAppViewRoutes = (useExtendedRoutes: boolean = false): RouteInter
                 path: AppConstants.getPaths().get("USERSTORES"),
                 protected: true,
                 showOnSidePanel: true
+            },
+            {
+                category: "extensions:manage.sidePanel.categories.userManagement",
+                component: lazy(() => import("../../provisioning/pages/outbound-provisioning-settings")),
+                exact: true,
+                icon: {
+                    icon: <UserGroupIcon className="icon" fill="black" />
```
Comment (savindi7) [inline apps/console/src/features/core/configs/routes.tsx:1689]: aren't we using the same icon for groups?

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+export const OutboundProvisioningConnectorDeleteWizard: FunctionComponent<
+    OutboundProvisioningConnectorDeleteWizardPropsInterface
+> = (props: OutboundProvisioningConnectorDeleteWizardPropsInterface): ReactElement => {
+
+    const {
+        isOpen,
+        onClose,
+        deletingIdp,
+        onConfirm,
+        [ "data-componentid" ]: testId
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx:50]:
```suggestion
        [ "data-componentid" ]: componentId
```

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                            ".deleteOutboundProvisioningIDP.assertionHint"
+                        }
+                        tOptions={ { name: deletingIdp?.idp } }
+                    >
+                        Please type <strong>{ deletingIdp?.idp }</strong> to confirm.
+                    </Trans>
+                </p>
+            ) }
+            assertionType="input"
+            primaryAction="Confirm"
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx:75]: Shouldn't we i18n the action text?

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+
+    const {
+        initialValues,
+        idpList,
+        triggerSubmit,
+        onSubmit,
+        isEdit,
+        readOnly,
+        isSubmitting,
+        [ "data-componentid" ]: testId
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx:75]:
```suggestion
        [ "data-componentid" ]: componentId
```

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+            blocking: isBlockingChecked ? isBlockingChecked : !!values.get("blocking").includes("blocking"),
+            connector: connector,
+            idp: idpName,
+            jit: isJITChecked ? isJITChecked : !!values.get("jit").includes("jit"),
+            rules: isRulesChecked ? isRulesChecked : !!values.get("blocking").includes("blocking")
+        };
+    };
+
+    return (
+        <Forms
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx:190]: Is it intentional to use the deprecated forms module?

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-wizard.tsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { useTrigger } from "@wso2is/forms";
+import { Heading, LinkButton, PrimaryButton, Steps } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement } from "react";
+import { useTranslation } from "react-i18next";
+import { Grid, Modal } from "semantic-ui-react";
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-wizard.tsx:24]: Shall we use Grid from oxygen UI?

File: apps/console/src/features/core/configs/routes.tsx [hunk 9]
```diff
@@ -1680,6 +1680,20 @@ export const getAppViewRoutes = (useExtendedRoutes: boolean = false): RouteInter
...
+                },
+                id: "outboundProvisioningSettings",
```
Comment (pavinduLakshan) [inline apps/console/src/features/core/configs/routes.tsx:1691]: Have the CRUD scopes for this route id already been added to deployment.config.json?

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx (hunk 7 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx:190]: Due to the urgency of the feature, an existing codebase has been used here. As few modifications for props were needed, couldn't reuse the existing component itself. Had to duplicate the codebase and do relevant modifications.

Comment (pavinduLakshan) [reply apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx:190]: Ack

File: apps/console/src/features/core/configs/routes.tsx (hunk 3 above)
Comment (JayaShakthi97) [reply apps/console/src/features/core/configs/routes.tsx]: Updated with b2b26abeb4ea42fd52d73cd016be02dc68af453f

File: apps/console/src/features/provisioning/api/outbound-provisioning.ts (hunk 1 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/api/outbound-provisioning.ts]: Fixed with b2b26abeb4ea42fd52d73cd016be02dc68af453f

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx (hunk 4 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx]: Fixed with b2b26abeb4ea42fd52d73cd016be02dc68af453f

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx (hunk 6 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx]: Fixed with b2b26abeb4ea42fd52d73cd016be02dc68af453f

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx (hunk 5 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx]: Fixed with 8246d33c5a99b550fe7ac283a1b5f721f2980628

File: apps/console/src/features/provisioning/api/outbound-provisioning.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+export const updateResidentApplicationOutboundProvisioningList = (
+    outboundProvisioningList: OutboundProvisioningConfigurationInterface[]
+): Promise<any> => {
+    const requestConfig: AxiosRequestConfig = {
+        data: {
+            outboundProvisioningIdps: outboundProvisioningList
+        },
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost,
```
Comment (pavinduLakshan) [inline apps/console/src/features/provisioning/api/outbound-provisioning.ts:40]: Does it make sense to send this header in request, as Access-Control-Allow-Origin is a response header [1]?

[1] https://stackoverflow.com/a/10636765/8810941

File: modules/i18n/src/translations/si-LK/portals/console.ts [hunk 11]
```diff
@@ -3482,6 +3482,71 @@ export const console: ConsoleNS = {
                         }
                     }
                 },
+                resident: {
+                    provisioning: {
+                        outbound: {
+                            actions: {
+                                addIdp: "නව කොමිෂන් සභා"
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/si-LK/portals/console.ts:3489]: Ref: https://translate.google.com/?sl=en&tl=si&text=New%20Provisioner&op=translate&hl=en

```suggestion
                                addIdp: "නව සැපයුම්කරු"
```

File: modules/i18n/src/translations/si-LK/portals/console.ts [hunk 12]
```diff
@@ -3482,6 +3482,71 @@ export const console: ConsoleNS = {
...
+                                        description: "පිටතට යන ප්රතිපාදන වින්යාසය යාවත්කාලීන කිරීමේදී යමක් වැරදී ඇත.",
+                                        message: "යාවත්කාලීන දෝෂය"
+                                    },
+                                    success: {
+                                        description: "පිටතට යන ප්රතිපාදන වින්යාසය සාර්ථකව යාවත්කාලීන කරන ලදි.",
+                                        message: "යාවත්කාලීන කිරීම සාර්ථකයි"
+                                    }
+                                }
+                            },
+                            subHeading: "නේවාසික අයදුම්පත සඳහා පිටතට යන ප්රතිපාදන සැකසුම් වින්යාස කරන්න."
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/si-LK/portals/console.ts:3546]:
```suggestion
                            subHeading: "නේවාසික යෙදුම සඳහා පිටතට යන ප්රතිපාදන සැකසුම් වින්යාස කරන්න."
```

File: apps/console/src/features/provisioning/api/outbound-provisioning.ts (hunk 10 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/api/outbound-provisioning.ts]: Addressed 4c98502d24cc3d1a6a564b3468bc1768fd727766

File: apps/console/src/features/core/configs/routes.tsx (hunk 9 above)
Comment (JayaShakthi97) [reply apps/console/src/features/core/configs/routes.tsx:1713]: Fixed 8fdd53d3d07b76a7666f7d305cf20677598271ba

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+                        Please type <strong>{ deletingIdp?.idp }</strong> to confirm.
+                    </Trans>
+                </p>
+            ) }
+            assertionType="input"
+            primaryAction={ t("common:confirm") }
+            secondaryAction={ t("common:cancel") }
+            onSecondaryActionClick={ onClose }
+            onPrimaryActionClick={ () => onConfirm(deletingIdp) }
+            data-testid={ `${ componentId }-connector-delete-confirmation-modal` }
```
Comment (Achintha444) [inline apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx:79]: add `data-componentid`. Check other places

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx [hunk 14]
```diff
@@ -0,0 +1,351 @@
...
+import { Grid } from "semantic-ui-react";
+import { OutboundProvisioningConnectorInterface } from "../../connections/models/connection";
+import { getIdentityProviderDetail } from "../../identity-providers/api/identity-provider";
+import { IdentityProviderInterface } from "../../identity-providers/models/identity-provider";
+
+/**
+ * Proptypes for the outbound provisioning connector setup form component.
+ */
+interface OutboundProvisioningConnectorSetupFormPropsInterface extends IdentifiableComponentInterface {
+    initialValues: any;
```
Comment (Achintha444) [inline apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx:34]: refrain from using `any` if possible

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx (hunk 13 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/components/outbound-provisioning-connector-delete-wizard.tsx]: Fixed with 0d107596adbd4c986a3cb65b0f0f76f6662870fa

File: apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx (hunk 14 above)
Comment (JayaShakthi97) [reply apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx]: Fixed with 0d107596adbd4c986a3cb65b0f0f76f6662870fa

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**